app/controllers/recipes_controller.rb

```diff
@@ -21,6 +21,36 @@ def create
     end
   end
 
+  def edit
+    if current_user.id == Recipe.find(params[:id]).user.id
+      @recipe = current_user.recipes.find(params[:id])
+    else
+      flash[:error] = "You are not authorized to edit this recipe."
+      redirect_to root_path
+    end
+  end
+
+  def update
+    if current_user.id == Recipe.find(params[:id]).user.id
+      @recipe = current_user.recipes.find(params[:id])
+      if @recipe.update_attributes(recipe_params)
+        flash[:notice] = "Recipe updated."
+        redirect_to @recipe
+      else
+        error_list = "<ul>"
```

Comment:
Try to avoid building up HTML strings in the controller, and aim to keep HTML in the view templates and view helpers where possible.

The `@recipe.errors.full_messages` array can be accessed in the edit template to build up an HTML list. The `@recipe.valid?` method may also be useful.